### ISSUE TITLE
ConditionalOnWebApplication not found after upgrading to 2.9.0

### ISSUE THREAD

**Issue by gionn** [springfox/springfox#2481]
I have a classic Spring MVC application (4.3.14.RELEASE) and after upgrading from 2.8.0 to 2.9.0 I get:

```
[ERROR] com.sun.tools.javac.code.Symbol$CompletionFailure: class file for org.springframework.boot.autoconfigure.condition.ConditionalOnWebApplication not found
```

To workaround the issue, I created an empty annotation class into my classpath:

```
package org.springframework.boot.autoconfigure.condition;

public @interface ConditionalOnWebApplication
{
}
```

**Comment by dilipkrish** [comment]
Thanks for reporting!

**Comment by dilipkrish** [comment]
Could you try the lates 2.9.1-SNAPSHOT and see if it fixes it

**Comment by gionn** [comment]
CI says green! :1st_place_medal:

**Comment by dilipkrish** [comment]
Fixed via #2438

**Comment by cablegunmaster** [comment]
Can there be a check in the whole project for import org.springframework.boot.* ? 
As there are annotations used which uses spring boot and im just using spring.

For example in springfox-swagger2-2.9.2 I found  @ConditionalOnWebApplication

``
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'swagger2ControllerMapping' defined in class path resource [springfox/documentation/swagger2/configuration/Swagger2DocumentationConfiguration.class]: Unsatisfied dependency expressed through method 'swagger2ControllerMapping' parameter 2; nested exception is org.springframework.beans.factory.NoSuchBeanDefinitionException: No qualifying bean of type 'springfox.documentation.swagger2.mappers.ServiceModelToSwagger2Mapper' available: expected at least 1 bean which qualifies as autowire candidate. Dependency annotations: {}
		at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:787)
		at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:528)
		at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1338)
		at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1177)
		at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:557)
		at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:517)
		at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:323)
		at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
		at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:321)
		at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:202)
		at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:879)
		at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:878)
		at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:550)
		at org.springframework.web.context.ContextLoader.configureAndRefreshWebApplicationContext(ContextLoader.java:401)
		at org.springframework.web.context.ContextLoader.initWebApplicationContext(ContextLoader.java:292)
		at org.springframework.web.context.ContextLoaderListener.contextInitialized(ContextLoaderListener.java:103)
		at org.apache.catalina.core.StandardContext.listenerStart(StandardContext.java:4683)
		at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5146)
		at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
		at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:717)
		at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:690)
		at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
``
